trio/_util.py

```diff
@@ -216,7 +216,9 @@ def __new__(cls, name, bases, cls_namespace):
         for base in bases:
             if isinstance(base, Final):
                 raise TypeError(
-                    "`%s` does not support subclassing" % base.__name__
+                    "the {!r} class does not support subclassing".format(
+                        base.__name__
```

Comment:
Do we want to do `{__module__}.{__qualname__}` for maximum comprehensibility?

Reply:
Good idea! Done.